OpenMarket/OpenMarket/Application/Presentation/ViewController/MarketProductsViewController.swift

```diff
@@ -0,0 +1,193 @@
+            self.gridCollectionView?.isHidden = !self.listCollectionView.isHidden
+        }
+    }
+    
+    private var productsModel: [ProductEntity] = []
+    
+    private var listCollectionView: UICollectionView!
+    private var listDataSource: UICollectionViewDiffableDataSource<Section, ProductEntity>!
+    
+    private var gridCollectionView: UICollectionView!
+    private var gridDataSource: UICollectionViewDiffableDataSource<Section, ProductEntity>!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        
+        setNavigationItems()
+        fetchData()
+    }
+}
+
+
+private extension MarketProductsViewController {
+    func fetchData() {
+        let url = "https://market-training.yagom-academy.kr/api/products?page_no=1&items_per_page=50"
+        
+        let openMarket = NetworkProvider(session: URLSession.shared)
+        openMarket.requestAndDecode(url: url, dataType: ProductList.self) { result in
+            switch result {
+            case .success(let productList):
+                productList.pages.forEach { product in
+                    let item = ProductEntity(thumbnailImage: product.thumbnailImage!, name: product.name, currency: product.currency, originalPrice: product.price, discountedPrice: product.bargainPrice, stock: product.stock)
+                    self.productsModel.append(item)
+                }
+                
+                DispatchQueue.main.async {
+                    self.createGridCollectionView()
+                    self.configureGridDataSource()
+                    self.gridCollectionView.isHidden = true
+                    self.createListCollectionView()
+                    self.configureListDataSource()
+                }
+                
+            case .failure(let error):
+                self.presentConfirmAlert(message: error.errorDescription ?? "")
+            }
+        }
+    }
+    
+    func presentConfirmAlert(message: String) {
+        let alertController = UIAlertController(title: "알림", message: message, preferredStyle: .alert)
+        let confirmAction = UIAlertAction(title: "확인", style: .default, handler: nil)
+        alertController.addAction(confirmAction)
+        self.present(alertController, animated: false)
+    }
+    
+    @objc func didSegmentedControlTapped(_ segment: UISegmentedControl) {
+        self.shouldHideListView = segment.selectedSegmentIndex != 0
+    }
+    
+    func setNavigationItems() {
+        self.navigationItem.titleView = self.segmentedControl
+        self.navigationItem.rightBarButtonItem  = UIBarButtonItem(title: "+", style: .plain, target: self, action: #selector(addTapped))
+        self.segmentedControl.addTarget(self, action: #selector(didSegmentedControlTapped(_:)), for: .valueChanged)
+        self.segmentedControl.selectedSegmentIndex = 0
+    }
+    
+    @objc func addTapped() {
+        
+    }
```

Comment:
```suggestion
    @objc func addTapped() {
         #warning("추후 구현 예정입니다") 
    }
```
=> 이렇게 표시해주는 것이 좋습니다.